src/Features/Core/Portable/AddConstructorParametersFromMembers/AddConstructorParametersFromMembersCodeRefactoringProvider.cs

```diff
@@ -54,24 +54,58 @@ public async Task<ImmutableArray<CodeAction>> AddConstructorParametersFromMember
             }
         }
 
+        /// <summary>
+        /// Try to find a constructor in <paramref name="containingType"/> whose parameters is the subset of <paramref name="parameters"/> by comparing name.
+        /// If multiple constructors meet the condition, the one with more parameters will be returned.
+        /// It will not consider those constructors as potential candidates
+        /// 1. Constructor with empty parameter list.
+        /// 2. Constructor's parameter list contains 'ref' or 'params'
+        /// </summary>
+        protected override IMethodSymbol GetDelegatedConstructor(
+            INamedTypeSymbol containingType,
+            ImmutableArray<IParameterSymbol> parameters)
+        {
+            return containingType.InstanceConstructors
+                .WhereAsArray(constructor => IsParamtersContainedInConstructor(constructor, parameters.SelectAsArray(p => p.Name)))
+                .OrderByDescending(constructor => constructor.Parameters.Length)
+                .FirstOrDefault();
+        }
+
+        private bool IsParamtersContainedInConstructor(
+            IMethodSymbol constructor,
+            ImmutableArray<string> parametersName)
+        {
+            var constructorParams = constructor.Parameters;
+            if (constructorParams.Length > 0
+                && constructorParams.All(parameter => parameter.RefKind == RefKind.None)
+                && !constructorParams.Any(p => p.IsParams))
+            {
+                return parametersName.Except(constructorParams.Select(p => p.Name)).Any();
```

Comment:
how does this work for VB/C# and properties that might be capitalized?  We should probably be case-insensitive here.

Reply:
This part contains some previous logic of this code if you take a look at the method `DetermineParameters()` in this class.  It works like this
User select some members ----> Their name is converted to CamelCase (The case you mentioned, property, if its first letter is Upper case like `int HelloWorld`, it will be converted to `helloWorld`) ----> And then here we are using `helloWorld` to compare with the parameters in constructor. Then finally an assignment statement `this.HelloWorld = helloWorld` will be added to constructor.

Reply:
I found that the previous unit tests only focus on field, I will add some Property tests